AI/daily_data_collection/main.py

```
@@ -0,0 +1,1165 @@
+    return datetime.now(KST).date()
+
+
+def get_last_date_in_table(db_name: str, table: str, date_col: str) -> Optional[datetime.date]:
+    """
+    테이블의 날짜 컬럼(date_col)에서 MAX(date)를 얻는 함수
+    """
+    from sqlalchemy import text
+    engine = get_engine(db_name)
+
+    with engine.connect() as conn:
+        res = conn.execute(text(f"SELECT MAX({date_col}) FROM {table};")).scalar()
```

Ruff (0.14.6)

Possible SQL injection vector through string-based query construction

(S608)